cartographer_grpc/map_builder_server.cc

```diff
@@ -34,26 +34,39 @@ const cartographer::common::Duration kPopTimeout =
 }  // namespace
 
 MapBuilderServer::MapBuilderContext::MapBuilderContext(
-    cartographer::mapping::MapBuilder* map_builder,
-    cartographer::common::BlockingQueue<std::unique_ptr<SensorData>>*
-        sensor_data_queue)
-    : map_builder_(map_builder), sensor_data_queue_(sensor_data_queue) {}
+    MapBuilderServer* map_builder_server)
+    : map_builder_server_(map_builder_server) {}
 
 cartographer::mapping::MapBuilder&
 MapBuilderServer::MapBuilderContext::map_builder() {
-  return *map_builder_;
+  return map_builder_server_->map_builder_;
 }
 
 cartographer::common::BlockingQueue<
     std::unique_ptr<MapBuilderServer::SensorData>>&
 MapBuilderServer::MapBuilderContext::sensor_data_queue() {
-  return *sensor_data_queue_;
+  return map_builder_server_->sensor_data_queue_;
+}
+
+cartographer::mapping::TrajectoryBuilderInterface::LocalSlamResultCallback
+MapBuilderServer::MapBuilderContext::local_slam_result_callback() {
```

Comment:
Rename to GetLocalSlamResultCallbackForSubscriptions? It behaves like this.

Reply:
Done.